### PR TITLE
Modernize EventWithHistoryEDFilter

### PR DESCRIPTION
**Pull request by makortel**
#### PR description:

In the context of #25090, this PR modernizes EventWithHistoryEDFilter by
- Move to `edm::global` module type
  * The contained `EventWithHistory` objects appear to be thread-safe by visual inspection and by not being flagged by the static analyzer
- Remove empty methods
- Use range-for
- Use trailing underscore for member data following the code guidelines

#### PR validation:

Code compiles.

### CODE REVIEW

**Comment by makortel** [comment]
@cmsbuild, please test

File: DPGAnalysis/SiStripTools/plugins/EventWithHistoryEDFilter.cc [hunk 1]
```diff
@@ -40,20 +40,17 @@
...
-  bool filter(edm::Event&, const edm::EventSetup&) override;
-  void endJob() override;
+  bool filter(edm::StreamID streamId, edm::Event&, const edm::EventSetup&) const override;
 
   // ----------member data ---------------------------
 
-  std::vector<EventWithHistoryFilter> _ehfilters;
-  bool _debu;
+  std::vector<EventWithHistoryFilter> ehfilters_;
+  bool debu_;
```
Comment (Dr15Jones) [inline DPGAnalysis/SiStripTools/plugins/EventWithHistoryEDFilter.cc]:
```suggestion
  const bool debu_;
```

**Comment by makortel** [comment]
@cmsbuild, please test

**Comment by fabiocos** [comment]
+1

**Comment by fabiocos** [comment]
merge

**Comment by santocch** [comment]
+1

**Pull request merged**